system-helper/flatpak-system-helper.c

```diff
@@ -221,6 +825,40 @@ handle_deploy (FlatpakSystemHelper   *object,
       return TRUE;
     }
 
+  ongoing_pull = take_ongoing_pull_by_dir (arg_repo_path);
+  if (ongoing_pull != NULL)
+    {
+      g_autofree gchar *bindfs_repo_path = NULL;
+      g_autofree gchar *real_repo_path = NULL;
+
+      if (umount (ongoing_pull->user_dir_name) == -1)
+        {
+          ongoing_pull_teardown (ongoing_pull);
+          glnx_throw_errno_prefix (&error, "Unmounting %s failed", ongoing_pull->user_dir_name);
+          g_dbus_method_invocation_return_gerror (invocation, error);
+          return TRUE;
+        }
+
+      if (ongoing_pull->bindfs)
+        {
+          g_subprocess_wait (ongoing_pull->bindfs, NULL, NULL);
```

Comment:
The unmount can succeed, yet the fuse daemon doesn't exist. For instance, if the mount lives on in another namespace. This means this could block potentially forever. We should probably have some kind of timeout here.